modules/storages/app/controllers/storages/admin/projects_storages_controller.rb

```diff
@@ -107,6 +109,11 @@ def edit
     # @project_storage is used in the view in order to render the form for a new object
     @project_storage = @object
 
+    @last_project_folders = Storages::LastProjectFolder
+                              .where(projects_storage: @project_storage)
+                              .index_by(&:mode)
+                              .transform_values(&:origin_folder_id)
```

Comment:
The following should return the same result, but we fetch less data from database and it is more concise imho.
```suggestion
    @last_project_folders = Storages::LastProjectFolder
                              .where(projects_storage: @project_storage)
                              .pluck(:mode, :origin_folder_id)
                              .to_h
```

Reply:
True, it is almost the same SQL:

```sql
SELECT "last_project_folders".* FROM "last_project_folders" WHERE "last_project_folders"."projects_storage_id" = $1  [["projects_storage_id", 23]]

SELECT "last_project_folders"."mode", "last_project_folders"."origin_folder_id" FROM "last_project_folders" WHERE "last_project_folders"."projects_storage_id" = $1  [["projects_storage_id", 23]]
```

Reply:
will take your suggestion.